commit a10473c752b8aeb945c7b551560172038ccb4848
Author: Dan Rosenberg <drosenberg@vsecurity.com>
Date:   Wed Sep 15 11:43:12 2010 +0000

    drivers/net/cxgb3/cxgb3_main.c: prevent reading uninitialized stack memory
    
    Fixed formatting (tabs and line breaks).
    
    The CHELSIO_GET_QSET_NUM device ioctl allows unprivileged users to read
    4 bytes of uninitialized stack memory, because the "addr" member of the
    ch_reg struct declared on the stack in cxgb_extension_ioctl() is not
    altered or zeroed before being copied back to the user.  This patch
    takes care of it.
    
    Signed-off-by: Dan Rosenberg <dan.j.rosenberg@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/cxgb3/cxgb3_main.c b/drivers/net/cxgb3/cxgb3_main.c
index 3a31272..95f913e 100644
--- a/drivers/net/cxgb3/cxgb3_main.c
+++ b/drivers/net/cxgb3/cxgb3_main.c
@@ -1890,6 +1890,8 @@ static int cxgb_extension_ioctl(struct net_device *dev, void __user *useraddr)
 	case CHELSIO_GET_QSET_NUM:{
 		struct ch_reg edata;
 
+		memset(&edata, 0, sizeof(struct ch_reg));
+
 		edata.cmd = CHELSIO_GET_QSET_NUM;
 		edata.val = pi->nqsets;
 		if (copy_to_user(useraddr, &edata, sizeof(edata)))
